util/panic_hook/src/lib.rs

```diff
@@ -24,16 +24,26 @@ use std::thread;
 use std::process;
 use backtrace::Backtrace;
 
-/// Set the panic hook
+/// Set the panic hook to write to stderr and abort the process when a panic happens.
 pub fn set_abort() {
-	set_with(|| process::abort());
+	set_with(|msg| {
+		let _ = io::stderr().write_all(msg.as_bytes());
+		process::abort()
+	});
 }
 
-/// Set the panic hook with a closure to be called afterwards.
-pub fn set_with<F: Fn() + Send + Sync + 'static>(f: F) {
+/// Set the panic hook with a closure to be called. The closure receives the panic message.
+///
+/// Depending on how Parity was compiled, after the closure either returns the process aborts or
+/// unwinding starts.
```

Comment:
I don't understand this, can you rephrase?

I guess it should be something like `"after the closure has been executed it either returns the exit code when the process terminates or the unwinding starts"`